daemon/cdi.go

```diff
@@ -0,0 +1,42 @@
+package daemon
+
+import (
+	"fmt"
+
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+// initializeCdi initializes the CDI registry and registers the CDI device driver.
+func initializeCdi(d *Daemon) error {
+	if !d.HasExperimental() {
+		return nil
+	}
+
+	return registerCdiDriver()
+}
+
+// injectDevices injects a set of CDI devices into the specified OCI specification.
+func injectDevices(s *specs.Spec, cdiDeviceNames []string) error {
+	if len(cdiDeviceNames) == 0 {
+		return nil
+	}
+
+	registry := cdi.GetRegistry()
+	if err := registry.Refresh(); err != nil {
+		// We don't consider registry refresh failure a fatal error.
+		// For instance, a dynamically generated invalid CDI Spec file for
+		// any particular vendor shouldn't prevent injection of devices of
+		// different vendors. CDI itself knows better and it will fail the
+		// injection if necessary.
+		logrus.Debugf("The following error was triggered when refreshing the CDI registry: %v", err)
```

Comment:
```suggestion
		logrus.WithError(err).Debug("Refreshing the CDI registry failed")
```

Reply:
Thanks. Not used to this use of `logrus`.